src/lib/cache/index.js

```diff
@@ -0,0 +1,87 @@
+import fs from "fs";
+import path from "path";
+
+import serverFn from "@/promisetracker/lib/server";
+
+const PROMISES_CACHE_FILENAME = "promises.json";
+const SITES_CACHE_FILENAME = "sites.json";
+
+// async function fetchAllNodes(url, options = { headers }, times = 0) {
+//   const response = await fetch(url, options);
+//   const resjson = await response.json();
+//   const data = resjson.results;
+//   if (resjson.next) {
+//     const nextData = await fetchAllNodes(resjson.next, options, times + 1);
+//     return { ...nextData, results: data.concat(nextData.results) };
+//   }
+
+//   return { ...resjson, results: data };
+// }
+
+// async function fetchNodes(days = 7) {
+//   const toDate = new Date();
+//   const fromDate = toDate.setDate(toDate.getDate() - days);
+//   const lastNotify = new Date(fromDate).toISOString();
+
+//   return fetchAllNodes(
+//     `https://api.sensors.africa/v1/node?last_notify__gte=${lastNotify}`
+//   );
+// }
+
+function cache(siteSlug, { fetchFor }) {
+  const server = serverFn(siteSlug);
+  const CACHE_TIMEOUT = server.env("CACHE_TIMEOUT");
+
+  async function read(filename) {
+    const publicDirectory = path.join(process.cwd(), "public/data");
```

Comment:
not a major thing but seems like we can have these in their own function to remove the repetitions
```
const publicDirectory = path.join(process.cwd(), "public/data");
const filePath = path.join(publicDirectory, filename);
```